tensorwaves/physics/helicityformalism/kinematics.py

```diff
@@ -1,16 +1,262 @@
+r"""Kinematic based calculations for the helicity formalism.
 
-class Kinematics():
-  def __init__(self):
-    pass
+It's responsibilities are defined by the interface
+:class:`.interfaces.Kinematics`.
 
-  def register_invariant_mass(self):
-    pass
+Here, the main responsibility is the conversion of general kinematic
+information of a reaction to helicity formalism specific quantities
 
-  def register_helicity_angles(self):
-    pass
+:math:`(s, \theta, \phi)`
 
-  def register_subsystem(self):
-    pass
+The basic building blocks are the :class:`~HelicityKinematics` and
+:class:`~SubSystem`.
+"""
 
-  def convert(events):
-    pass
\ No newline at end of file
+import logging
+from typing import Union
+
+import amplitf.kinematics as tfa_kin
+
+import numpy as np
+
+from tensorwaves.interfaces import Kinematics
+
+
+class SubSystem():
+    """Represents a part of a decay chain.
+
+    A SubSystem resembles a decaying state and its ingoing and outgoing state.
+    It is uniquely defined by
+
+    * :attr:`final_states`
+    * :attr:`recoil_state`
+    * :attr:`parent_recoil_state`
+    """
+
+    def __init__(self, final_states, recoil_state, parent_recoil_state):
+        """Fully initializes the :class:`~SubSystem`.
+
+        Args:
+            final_states: `tuple` of `tuple` s containing unique ids.
+                Represents the final state content of the decay products.
+            recoil_state: `tuple` of unique ids representing the recoil partner
+                of the decaying state.
+            parent_recoil_state: `tuple` of unique ids representing the recoil
+                partner of the parent state.
+        """
+        self._final_states = tuple(tuple(x) for x in final_states)
+        self._recoil_state = tuple(recoil_state)
+        self._parent_recoil_state = tuple(parent_recoil_state)
+
+    @property
+    def final_states(self):
+        """Get final state content of the decay products."""
+        return self._final_states
+
+    @property
+    def recoil_state(self):
+        """Get final state content of the recoil partner."""
+        return self._recoil_state
+
+    @property
+    def parent_recoil_state(self):
+        """Get final state content of the recoil partner of the parent."""
+        return self._parent_recoil_state
+
+    def __eq__(self, other):
```

Comment:
Currently, only the special methods `__call__` and `__init__` are rendered in the API.

Reply:
Hmm, I think we should add this since its part of the public API. A user can do `SubSystem1 == SubSystem2`

Reply:
It can be added here
https://github.com/ComPWA/tensorwaves/blob/4a06430ab9d070e0c879132c212cf42f854dadef/doc/conf.py#L62

Reply:
Ok I will add that

Reply:
See https://github.com/ComPWA/tensorwaves/pull/39/commits/8cb12e031798e40cbfb6a3a094e8dca7f91c3ce4